tools/test_containers_compose

```diff
@@ -2,11 +2,54 @@
 
 set -euo pipefail
 
-for i in webui worker; do
+wait_until(){
+  command=$1
+  timeout=$2
+
+  i=1
+  until eval "$command"
+  do
+      ((i++))
+      if [[ $i -gt $timeout ]]; then
+          echo "Command '$command' aborted because of timeout"
+          exit 1
+      fi
+      sleep 1
+  done
+}
+
+setup_containers(){
   (
-    cd container/$i/ &&
-     for retry in {2..0}; do sudo docker-compose build && break; echo "Remaining retries $retry"; done &&
-    sudo docker-compose up -d &&
+    for retry in {2..0}; do sudo docker-compose build && break; echo "Remaining retries $retry"; done &&
+    sudo MOJO_CLIENT_DEBUG=1 docker-compose up -d &&
     (docker-compose ps --services --filter status=stopped | grep "^[[:space:]]*$") || (docker-compose logs; sudo docker-compose ps; exit 1)
   ) || exit 1;
-done
+}
+
+test_webui(){
+  (
+    workspace=$(mktemp -d) &&
+    trap 'docker-compose down; sudo rm -rf "$workspace"' EXIT &&
```

Comment:
Somehow I'm not comfortable running `sudo rm -rf` on an path which is the unvalidated output of another command. Are the `sudo` and the `-f` flag really necessary?

Reply:
Sudo yes because the workdir is created by docker-compose that is executed by sudo.

Reply:
works without force (-f)